shopify-client/src/components/Header.vue

```diff
@@ -35,9 +39,17 @@
 <script>
 export default {
   name: "Header",
+  methods:{
+    submitSearch: function(){
+      let keyword = "";
+      if(this.keyword){
+        keyword = "?keyword=" + this.keyword;
+      }
+      window.location.href = '/app/shops/List' + keyword;
```

Comment:
I believe this would refresh the page every time the user submits the search bar but doesn't enter any text.

Could try something like this:
```
if (this.keyword) {
    window.location.href = '/app/shops/List?keyword=' + this.keyword;
}
```

Reply:
That was intended to occur
No text would come up as undefined.
The url would take the literally name "undefined" and set it as '/app/shops/List?keyword=undefined'

Reply:
Empty string is considered `falsy` in JavaScript:
https://developer.mozilla.org/en-US/docs/Glossary/Falsy

So if the string was empty, the if statement would not pass.

I was saying in your original implementation, every time the user hits enter the page would redirect even if the user didn't enter anything.

Reply:
yes that was intended
I'm assuming that you want to remove that option

Reply:
Okay I see, this is so the user could go to the page containing all shops via the search bar?